typescript/lib/constructor_calls.ts

```diff
@@ -0,0 +1,219 @@
+        const result: StartServiceArgs = new StartServiceArgs();
+        result.setServiceId(String(serviceId));
+        result.setDockerImage(dockerImage);
+        const usedPortsMap: Map<string, boolean> = result.getUsedPortsMap();
+        for (let portId in usedPorts) {
+            usedPortsMap.set(portId, usedPorts[portId]);
+        }
+        const entrypointArgsArray: string[] = result.getEntrypointArgsList();
+        for (let entryPoint in entrypointArgs) {
+            entrypointArgsArray.push(entryPoint);
+        }
+        const cmdArgsArray: string[] = result.getCmdArgsList();
+        for (let cmdArg in cmdArgs) {
+            cmdArgsArray.push(cmdArg);
+        }
+        const dockerEnvVarArray: Map<string, string> = result.getDockerEnvVarsMap();
+        for (let dockerEnvId in dockerEnvVars) {
+            dockerEnvVarArray.set(dockerEnvId, dockerEnvVars[dockerEnvId]);
+        }
+        result.setEnclaveDataVolMntDirpath(enclaveDataVolMntDirpath);
+        const filesArtificatMountDirpathsMap: Map<string, string> = result.getFilesArtifactMountDirpathsMap();
+        for (let filesArtifactMountDirpathId in filesArtifactMountDirpaths) {
+            filesArtificatMountDirpathsMap.set(filesArtifactMountDirpathId, filesArtifactMountDirpaths[filesArtifactMountDirpathId]);
+        }
+
+        return result;
+}
+
+export function newGetServiceInfoArgs(serviceId: ServiceID): GetServiceInfoArgs{
+    const result: GetServiceInfoArgs = new GetServiceInfoArgs();
+    result.setServiceId(String(serviceId));
+
+    return result;
+}
+
+export function newRemoveServiceArgs(serviceId: ServiceID, containerStopTimeoutSeconds: number): RemoveServiceArgs {
+    const result: RemoveServiceArgs = new RemoveServiceArgs();
+    result.setServiceId(serviceId);
+    result.setContainerStopTimeoutSeconds(containerStopTimeoutSeconds);
+
+    return result;
+}
+
+export function newPartitionServices(serviceIdStrSet: Set<string>): PartitionServices{
+    const result: PartitionServices = new PartitionServices();
+    const partitionServicesMap: Map<string, boolean> = result.getServiceIdSetMap();
+    for (let serviceIdStr in serviceIdStrSet) {
+        partitionServicesMap.set(serviceIdStr, true);
+    }
+
+    return result;
+}
+
+export function newRepartitionArgs(
+    partitionServices: Map<string, PartitionServices>, 
+    partitionConns: Map<string, PartitionConnections>,
+    defaultConnection: PartitionConnectionInfo): RepartitionArgs {
+    const result: RepartitionArgs = new RepartitionArgs();
+    const partitionServicesMap: Map<PartitionID, Map<ServiceID, boolean>> = new Map();
+    for (let partitionServiceId in partitionServices) {
+        partitionServicesMap.set(partitionServiceId, partitionServices[partitionServiceId]);
+    };
+    const partitionConnsMap: Map<PartitionID, Map<PartitionID, PartitionConnectionInfo>> = new Map();
+    for (let partitionConnId in partitionConns) {
```

Comment:
Iterate with Map.entries(), here and everywhere else in this file that it's applicable